xpublish_edr/plugin.py

```diff
@@ -225,15 +224,82 @@ def get_area(
 
             logger.debug(f"Dataset projected to {query.crs}: {ds}")
 
-            if query.format:
+            if query.f:
+                try:
+                    format_fn = output_formats()[query.f]
+                except KeyError as e:
+                    logger.error(f"Error getting format function: {e}")
+                    raise HTTPException(
+                        404,
+                        f"{query.f} is not a valid format for EDR area queries. "
+                        "Get `./area/formats` for valid formats",
+                    )
+
+                return format_fn(ds)
+
+            return to_cf_covjson(ds)
+
+        @router.get("/cube", summary="Cube query")
+        def get_cube(
+            request: Request,
+            query: Annotated[EDRCubeQuery, Query()],
+            dataset: xr.Dataset = Depends(deps.dataset),
+        ):
+            """
+            Returns gridded cube data based on bbox coordinates and optional elevation
+
+            Extra selecting/slicing parameters can be provided as extra query parameters
+            """
+            try:
+                ds = query.select(dataset, dict(request.query_params))
+            except ValueError as e:
+                logger.error(f"Error selecting from query while selecting by cube: {e}")
+                raise HTTPException(
+                    status_code=404,
+                    detail=f"Error selecting from query: {e.args[0]}",
+                )
+
+            logger.debug(f"Dataset filtered by query params {ds}")
+
+            try:
+                ds = select_by_bbox(ds, query.project_bbox(ds))
```

Comment:
If we're moving to geometry specific queries, should we move more of the selection logic into them? 

That way instead of having `query.select` and `select_by_bbox` in the `router.get` methods, we have a single `query.select` that includes the geometry specific selection? I would also help clean up & shorten the methods in `plugin.py`.

Reply:
I think some cleanup needs to b e done here but i want to leave it for a follow up